examples/CMakeLists.txt

```diff
@@ -46,17 +46,39 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
     PATTERN "DTMB-5415_bulbous_bow.iges"  # step-54
     )
 
-  MESSAGE(STATUS "Setting up examples - Done")
-ENDIF()
+  #
+  # Set up all executables:
+  #
+  FILE(GLOB _steps ${CMAKE_CURRENT_SOURCE_DIR}/step-*/*.cc)
+  FOREACH(_step ${_steps})
+    GET_FILENAME_COMPONENT(_name ${_step} NAME_WE)
+    GET_FILENAME_COMPONENT(_directory ${_step} DIRECTORY)
+    SET(_set_up TRUE)
+    IF(EXISTS "${_directory}/dependency")
+      FILE(READ "${_directory}/dependency" _dependency)
+      # I will so totally go to hell for that:
+      RUN_COMMAND("
+        IF(NOT (${_dependency}))
+          MESSAGE(STATUS \"  ${_name} - dependency not satisfied\")
+          SET(_set_up FALSE)
+        ENDIF()"
```

Comment:
Good, see you in hell :-D

It is a brutal trick to evaluate the `IF(NOT <expression>) ...` condition where the expression depends on the content of external files. This is necessary because cmake does not support "self modifying" cmake script.

I find the alternative of listing *every* example step with its configuration requirements in `examples/CMakeLists.txt` less appealing.